### PR TITLE
[Feature] Manual plan management

### LINKED ISSUE #800: [Feature] Manual plan management
- [ ] Plan add route
- [ ] Plan edit route

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #800
close with associatedemployers/igneous#344

image

image

image

image

image

### CODE REVIEW

File: app/components/benefits/manual-plan.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Component from '@glimmer/component';
+import { planForm, planOptions } from '../../config/forms/plans';
+import { inject as service } from '@ember/service';
+
+export default class BenefitsManualPlanComponent extends Component {
+currentForm = planForm
+planOptions = planOptions
+@service data
```
Comment (James1x0) [inline app/components/benefits/manual-plan.js:8]:
```suggestion
  currentForm = planForm
  planOptions = planOptions
  @service data
```

File: app/routes/account/benefits/plans/add.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class PlanAddRoute extends Route {
+  @service store
+
+  async model () {
+    return this.store.createRecord('plan', { fromService: 'manual' });
```
Comment (James1x0) [inline app/routes/account/benefits/plans/add.js:8]: Removing this, don't want anything in there for a manual plan
```suggestion

  async model () {
    return this.store.createRecord('plan');
```

File: app/templates/account/benefits/plans/add.hbs [hunk 3]
```diff
@@ -0,0 +1 @@
+<Benefits::ManualPlan @plan={{this.model}} @save={{this.save}} @cancel={{this.cancel}}/>
```
Comment (James1x0) [inline app/templates/account/benefits/plans/add.hbs:1]:
```suggestion
<Benefits::ManualPlan @plan={{this.model}} @save={{this.save}} @cancel={{this.cancel}}/>
```
Why is this a separate component?

File: app/templates/components/benefits/plan.hbs [hunk 4]
```diff
@@ -82,4 +82,13 @@
     </div>
     {{/if}}
   </div>
+  {{#if (eq @plan.fromService "manual") }}
```
Comment (James1x0) [inline app/templates/components/benefits/plan.hbs:85]: Change this logic to remove "from service". If it's not "fromService" it's a manual plan

File: app/templates/account/benefits.hbs [hunk 5]
```diff
@@ -17,9 +17,9 @@
             <a href="benefits/plans/integrations" id="action-item__action--delay" class="item">
               <i class="ellipsis horizontal icon"></i>link a carrier
```
Comment (James1x0) [inline app/templates/account/benefits.hbs:18]:
```suggestion
            <LinkTo @route="benefits.plans.integrations" id="action-item__action--delay" class="item">
              <i class="ellipsis horizontal icon"></i> Link a carrier
```

File: app/templates/account/benefits.hbs [hunk 6]
```diff
@@ -17,9 +17,9 @@
...
             </a>
```
Comment (James1x0) [inline app/templates/account/benefits.hbs:19]:
```suggestion
            </LinkTo>
```

File: app/templates/account/benefits.hbs [hunk 7]
```diff
@@ -14,12 +14,12 @@
           <i class="wrench icon"></i><i class="dropdown icon"></i>
           <div class="menu">
             <div class="header">Actions</div>
-            <a href="benefits/plans/integrations" id="action-item__action--delay" class="item">
-              <i class="ellipsis horizontal icon"></i>link a carrier
-            </a>
-            <a href="#" id="action-item__action--delay" class="item" {{!--{{on "click" add}} --}}>
-              <i class="ellipsis horizontal icon"></i>Manually add a plan
-            </a>
+            <LinkTo @route="benefits.plans.integrations" id="action-item__action--delay" class="item">
```
Comment (James1x0) [inline app/templates/account/benefits.hbs:17]:
```suggestion
            <LinkTo @route="account.benefits.plans.integrations" id="action-item__action--delay" class="item">
```

File: app/routes/account/benefits/plans/edit.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+export default class PlanEditRoute extends Route {
```
Comment (James1x0) [inline app/routes/account/benefits/plans/edit.js:3]:
```suggestion

export default class PlanEditRoute extends Route {
```

File: app/routes/account/benefits/plans/add.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+export default class PlanAddRoute extends Route {
+  @service store
+
+  async model () {
+    return this.store.createRecord('plan');
+  }
+
+  setupController (controller, model) {
+    controller.setProperties({ model });
+  }
```
Comment (James1x0) [inline app/routes/account/benefits/plans/add.js:13]:
```suggestion
```

File: app/components/quick-form.hbs [hunk 10]
```diff
@@ -1,7 +1,12 @@
 {{#let (element (if @tagName @tagName "form")) as |Tag|}}
   <Tag class="ui form" ...attributes>
     {{#each @form as |field|}}
-      {{#if (not (and field.displayIf (not (get @model field.displayIf))))}}
+      {{#if 
+        (not
+          (and field.displayIf (not (if field.displayIfEq (eq field.displayIfEq (get @model field.displayIf)) (get @model field.displayIf))))
+          (and field.displayIfNot (not-eq field.displayIfNot (get @model field.displayIfNot)) (get @model field.displayIfNot))
```
Comment (James1x0) [inline app/components/quick-form.hbs:7]: Does the not helper accept two args??

**Comment by James1x0** [comment]
A few things to review @JavaLavaMT

File: app/templates/account/benefits/plans/add.hbs (hunk 3 above)
Comment (JavaLavaMT) [reply app/templates/account/benefits/plans/add.hbs:1]: "Why is this a separate component?" not sure what you mean?

File: app/templates/components/benefits/plan.hbs (hunk 4 above)
Comment (JavaLavaMT) [reply app/templates/components/benefits/plan.hbs:85]: 🏁

File: app/components/quick-form.hbs (hunk 10 above)
Comment (JavaLavaMT) [reply app/components/quick-form.hbs:7]: not suer what I am missing. I have 2x `(and` in one `not` helper

Comment (James1x0) [reply app/components/quick-form.hbs:7]: Apparently according to their docs: https://github.com/jmurphyau/ember-truth-helpers they actually do accept a variable semaphore! Noice.

File: app/templates/account/benefits/plans/add.hbs (hunk 3 above)
Comment (James1x0) [reply app/templates/account/benefits/plans/add.hbs]: I thought this was a list item. Let's change the name to PlanForm or PlanEditForm.

**Comment by James1x0** [comment]
@JavaLavaMT Needs a resolution

**Pull request merged**